ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2964,7 +2964,52 @@ MmLoadSystemImage(IN PUNICODE_STRING FileName,
     PrefixName = *FileName;
 
     /* Check if we have a prefix */
-    if (NamePrefix) DPRINT1("Prefixed images are not yet supported!\n");
+    if (NamePrefix)
+    {
+        /* Check if "directory + prefix" is too long for the string */
+        Status = RtlUShortAdd(PrefixName.MaximumLength,
```

Comment:
Shouldn't we look at `PrefixName.Length` here only? (equivalently, `FileName->Length`)

Reply:
hmm, I might first thought I need to "enlarge" the buffer based on the buffer size of the file name. And yeah we're only copying the part of buffer that's been used by the string so using the buffer size is not necessary. I'll change it to Length later.